### DISCUSSION TITLE
Can't fix a bug because of the VBA linter

### DISCUSSION THREAD

**Discussion by 6DiegoDiego9** [sancarn/stdVBA#101] [General]
I just sent a pull request to fix a bug in stdArray. However, I get this completely unrelated error:

![image](https://github.com/sancarn/stdVBA/assets/109476941/cfd45b6b-71bb-45b0-8b44-a53ee9b762fb)

![image](https://github.com/sancarn/stdVBA/assets/109476941/297c318d-b59a-4765-a205-e0e2e9ae4369)

**Comment by sancarn** [comment]
Hey, you can ignore the VBA linter. It's a work in progress. I'll probably take it off this repo and leave it on a branch instead. Your PRs still come through to me :)

**Reply by DecimalTurn** [reply]
It seems to be the precompiler complaining about the use "é" in "Bézier" in that comment (further right) : 
https://github.com/sancarn/stdVBA/blob/15031262746c337d49e19cedba0da6aa9ad08f53/src/stdImage.cls#L63

I don't think it should complain however since VBA code authors should be allowed to add non-ascii characters in comments even if it might not display correctly on someone with a machine using a different text-encoding than the author on their machine.

Btw, is the precompiler useful for this project or is it just for testing and see what it can do? My understanding is that the precompiler and the VBA-Linter don't have to be used together (necessarly).

**Reply by sancarn** [reply]
> It seems to be the precompiler complaining about the use "é" in "Bézier" 

@beakerboy - one for you

> My understanding is that the precompiler and the VBA-Linter don't have to be used together.

Oh is that so? I had no idea honestly, I just copied what Beakerboy had in his repo 😅  Generally speaking there are still a ridiculous number of errors with whitespace etc. which I'm not really going to fix lol. So even the linter has very little affect on this project atm

**Reply by Beakerboy** [reply]
The precompiler is a separate project. All it does is take VBA code with Conditional Compilation statements as an input, and returns straight VBA as an output. If the code does not have CC statements, it is unnecessary. It’s also unnecessary if the CC statements only surround simple statements or complete blocks, i.e.  do not surround only PART of a control structure or function/sub declaration.

I’ll look at the non-ascii stuff. I have not implemented all the internationalization parts of the MS-VBA standard.

**Reply by sancarn** [reply]
> do not surround only PART of a control structure or function/sub declaration

Gotcha, guess that is where it will be required for some of stdVBA code.

**Reply by DecimalTurn** [reply]
I was still able to [run](https://github.com/DecimalTurn/stdVBA/actions/runs/9781907976/job/27007073144) the Linter on my fork without the precompilation step, so it might actually be unnecessary for this project atm. 

My understanding is that an example of code that would need to be precompiled would be like below where the declaration of the function is inside the CC statement, but not the `End Function` part.

```vba

#If VBA7 Then
	Private Function MyFunction(ByRef Value As LongPtr) As Boolean
#Else
	Private Function MyFunction(ByRef Value As Long) As Boolean
#End If

	[...]    

	End Function

```

@Beakerboy, please correct me if I'm wrong.

EDIT:
I actually found an example of that in the project, so my interpretation is clearly inaccurate 😅.
https://github.com/sancarn/stdVBA/blob/15031262746c337d49e19cedba0da6aa9ad08f53/src/stdClipboard.cls#L643-L653

**Reply by Beakerboy** [reply]
Your edit code snippet should not run with the linter. If it does, there’s a bug. The ANTLR grammar should complain about a missing `End Function`
